pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -758,18 +768,20 @@ private void writeMetadata()
           }
         }
 
+        Interval timeInterval =
+            new Interval(timeUnit.toMillis(startTime), timeUnit.toMillis(endTime), DateTimeZone.UTC);
+        boolean isValidTimeInterval = TimeUtils.isValidTimeInterval(timeInterval);
         if (!_config.isSkipTimeValueCheck()) {
```

Comment:
(minor) Suggest changing the check order a little bit and log a warning when time interval is not valid, but time value check is skipped
```suggestion
        if (isValidTimeInterval) {
          setSegmentTimeInterval(properties, startTime, endTime, timeUnit);
        } else {
          if (_config.isSkipTimeValueCheck()) {
            LOGGER.warn(...);
          } else {
            throw new IllegalStateException(...);
          }
        }
```